src/shared/sleep-config.c

```diff
@@ -108,77 +112,180 @@ int parse_sleep_config(SleepConfig **ret_sleep_config) {
+                return log_debug_errno(r, "Failed to initialize battery enumerator: %m");
+
+        FOREACH_DEVICE(e, dev) {
+                r = read_battery_capacity_percentage(dev);
+                if (r < 0) {
+                        log_device_debug_errno(dev, r, "Failed to get battery capacityi, ignoring: %m");
+                        continue;
+                }
+                if (r > BATTERY_LOW_CAPACITY_LEVEL)
+                        /* true only if battery capacity is above 5 */
+                        return true;
+        }
 
-        return r <= 5;
+        return false;
 }
 
-/* Battery percentage capacity fetched from capacity file and if in range 0-100 then returned */
-int read_battery_capacity_percentage(void) {
-        _cleanup_free_ char *bat_cap = NULL;
-        int battery_capacity, r;
+/* get sum of all batteries current charge level */
+int get_total_battery_capacity(void) {
+        _cleanup_(sd_device_enumerator_unrefp) sd_device_enumerator *e = NULL;
+        sd_device *dev;
+        int total_battery_capacity = 0, r;
 
-        r = read_one_line_file("/sys/class/power_supply/BAT0/capacity", &bat_cap);
-        if (r == -ENOENT)
-               return log_debug_errno(r, "/sys/class/power_supply/BAT0/capacity is unavailable. Assuming no battery exists: %m");
+        r = battery_enumerator_new(&e);
         if (r < 0)
-               return log_debug_errno(r, "Failed to read /sys/class/power_supply/BAT0/capacity: %m");
+                return log_debug_errno(r, "Failed to initialize battery enumerator: %m");
+
+        FOREACH_DEVICE(e, dev) {
+                r = read_battery_capacity_percentage(dev);
+                if (r < 0) {
+                        log_device_debug_errno(dev, r, "Failed to get battery capacityi, ignoring: %m");
+                        continue;
+                }
+                total_battery_capacity = total_battery_capacity + r;
+        }
+
+        return total_battery_capacity;
+}
+
+/* Store current capacity of each battery before suspension and timestamp */
+int hashmap_battery_name_with_battery_capacity(Hashmap **ret) {
+        _cleanup_(sd_device_enumerator_unrefp) sd_device_enumerator *e = NULL;
+        _cleanup_(hashmap_freep) Hashmap *current_capacity = NULL;
+        sd_device *dev;
+        int r;
+
+        assert(ret);
+
+        current_capacity = hashmap_new(&string_hash_ops_free);
+        if (!current_capacity)
+                return log_oom();
 
-        r = safe_atoi(bat_cap, &battery_capacity);
+        r = battery_enumerator_new(&e);
         if (r < 0)
-               return log_debug_errno(r, "Failed to parse battery capacity: %m");
+                return log_debug_errno(r, "Failed to initialize battery enumerator: %m");
 
-        if (battery_capacity < 0 || battery_capacity > 100)
-               return log_debug_errno(SYNTHETIC_ERRNO(ERANGE), "Invalid battery capacity");
+        FOREACH_DEVICE(e, dev) { /* For AC, the TYPE is mains */
```

Comment:
AC still not handled

Reply:
sonalis@cere:~/outreachyspring2022/test-sys  $ udevadm info /sys/class/power_supply/AC | grep POWER_SUPPLY_TYPE
E: POWER_SUPPLY_TYPE=Mains

As TYPE is Mains for AC, this is now handled. removing comment.